### PR TITLE
asciiquarium-transparent: init at unstable-2023-02-19

### PR DESCRIPTION
**Pull request by quantenzitrone**
## Description of changes

Asciiquarium with a few patches, notably background transparency

(use the --transparent flag to test it out)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/as/asciiquarium-transparent/package.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  makeWrapper,
+  perlPackages,
+  ...
```
Comment (wegank) [inline pkgs/by-name/as/asciiquarium-transparent/package.nix:7]:
```suggestion
```

File: pkgs/by-name/as/asciiquarium-transparent/package.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    rev = "653cd99a611080c776d18fc7991ae5dd924c72ce";
+    hash = "sha256-72LRFydbObFDXJllmlRjr5O8qjDqtlp3JunE3kwb5aU=";
+  };
+  nativeBuildInputs = [makeWrapper];
+  buildInputs = [perlPackages.perl];
+  installPhase = ''
+    mkdir -p $out/bin
+    cp asciiquarium $out/bin/asciiquarium
+    wrapProgram $out/bin/asciiquarium --set PERL5LIB ${perlPackages.makeFullPerlPath [perlPackages.TermAnimation]}
+  '';
```
Comment (wegank) [inline pkgs/by-name/as/asciiquarium-transparent/package.nix:24]:
```suggestion
  installPhase = ''
    runHook preInstall
    mkdir -p $out/bin
    cp asciiquarium $out/bin/asciiquarium
    wrapProgram $out/bin/asciiquarium --set PERL5LIB ${perlPackages.makeFullPerlPath [perlPackages.TermAnimation]}
    runHook postInstall
  '';
```

File: pkgs/by-name/as/asciiquarium-transparent/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  meta = with lib; {
+    description = mdDoc ''
```
Comment (wegank) [inline pkgs/by-name/as/asciiquarium-transparent/package.nix:26]:
```suggestion
    description = ''
```

Comment (Quantenzitrone) [reply pkgs/by-name/as/asciiquarium-transparent/package.nix:26]: why not mdDoc? aren't we supposed to migrate away from asciidoc?

Comment (wegank) [reply pkgs/by-name/as/asciiquarium-transparent/package.nix:26]: What you say is true for descriptions in modules, but `meta.description` in packages is in plain text, as far as I know.

Comment (wegank) [reply pkgs/by-name/as/asciiquarium-transparent/package.nix:26]: `mdDoc` is also a no-op, actually:

https://github.com/NixOS/nixpkgs/blob/558bcddd7b83ceb962cb4405a92f4060cb98cb3e/lib/options.nix#L346-L349

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 253439` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>asciiquarium-transparent</li>
  </ul>
</details>

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 253439` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>asciiquarium-transparent</li>
  </ul>
</details>

**Comment by Quantenzitrone** [comment]
thanks for reviewing and merging all my prs

**Pull request merged**